test_unstructured_ingest/expected-structured-output/confluence-diff/MFS/1605956.json

```diff
@@ -541,15 +530,15 @@
         "Insert"
       ],
       "emphasized_text_tags": [
-        "strong"
+        "b"
       ],
       "filetype": "text/html",
       "languages": [
         "eng"
       ]
     },
     "text": "Select Insert",
-    "type": "ListItem"
+    "type": "Title"
```

Comment:
@scanny - Do you know what caused these types to flip?

Reply:
This is an edge case we probably need to noodle a little bit. The HTML looks like this:
```html
<li>
  <p>Select <strong>Insert</strong></p>
</li>
```

**Current behavior**
- `<li>` is a block element. That means it starts a new paragraph->Element, as opposed to phrasing/inline content which just extends the paragraph it belongs to.
- `<p>` is _also_ a block element.
- A block element (like `<li>` in this case) generates one element for the text it contains and also one for each block-item child it contains.
- `<li>` contains 0 text and 1 block child so it generates the Element formed from the `<p>` element which in this case resolves to `Title`.

**Complication**
- It would be reasonable to say that because the `<p>` is "inside" the `<li>` that the element generated should be considered a `ListItem` instead of whatever type its text would classify to (`Title` in this case but could be `NarrativeText` etc.).
- This idea is given further credence by the fact this nested `<p>` text displays exactly that way in the browser; there is no separate paragraph "container", the `<p>` text just appears as the list item.
- The complication arises when the `<li>` contains multiple block elements or when it contains text as well as one or more block elements:
```html
<ul>
  <li>
    <p>Lorem</p>
    <p>Ipsum</p>
    <p>Dolor</p>
  </li>
  <li>Consectetur</li>
  <li>Adipiscing</li>
</ul>
```
- In the "contains multiple block-items" case, the browser displays the first block item as the list item and the remaining block items as "list-continuation" paragraphs as in this screenshot:
![Screenshot 2024-06-25 at 12 58 15 PM](https://github.com/Unstructured-IO/unstructured/assets/2062718/1c0a1399-a4b1-4d37-8a6f-f8f33c0f0bd6)
- However, there is no `ListContinuation` Element-type, so it's not immediately clear how to handle this conundrum.

**Conclusion**
Not having a good ready answer to the problem I went with the simplest thing that could possibly work and that was to leave the behavior as it defaults.

We could noodle it more or wait for feedback where it seems to matter for someone and think more about how to handle this case in the context of a concrete example.

Reply:
Btw, the old parser got this particular one right, but it was at the cost of squashing the entire contents of the `<li>` element into a single element, including when it contained a nested list: #3245 

That's the behavior that would squash a long list of nested list-items into a single ListItem element, like this example: https://github.com/Unstructured-IO/unstructured/pull/3218/files?show-deleted-files=true&show-viewed-files=true&file-filters%5B%5D=#diff-78e86d31815f4af1fe1cddeafa40432b9f5331fcd0385696ca81e1636cbdabf8L127

Original HTML:
```html
<ul id="subMenuNav">
  <li>
    <div class="subMenuNavList section-link">
      <a href="http://www.weather.gov/safetycampaign">Weather Safety</a>
    </div>

    <div class="sub">
      <ul>
        <li><a href="https://www.weather.gov/safety/airquality">Air Quality</a></li>
        <li><a href="https://www.weather.gov/safety/beachhazards"> Beach Hazards</a></li>
        <li><a href="https://www.weather.gov/safety/cold">Cold</a></li>
        <li><a href="https://www.weather.gov/safety/coldwater">Cold Water</a></li>
        <li><a href="https://www.weather.gov/safety/drought">Drought</a></li>
        <li><a href="https://www.weather.gov/safety/flood">Floods</a></li>
        <li><a href="https://www.weather.gov/safety/fog">Fog</a></li>
        <li><a href="https://www.weather.gov/safety/heat">Heat</a></li>
        <li><a href="https://www.weather.gov/safety/hurricane">Hurricanes</a></li>
        <li><a href="https://www.weather.gov/safety/lightning">Lightning Safety</a></li>
        <li><a href="https://www.weather.gov/safety/ripcurrent"> Rip Currents</a></li>
        <li><a href="https://www.weather.gov/safety/safeboating"> Safe Boating</a></li>
        <li><a href="https://www.weather.gov/safety/space">Space Weather</a></li>
        <li><a href="https://www.weather.gov/safety/heat-uv">Sun (Ultraviolet Radiation)</a></li>
        <li><a href="https://www.weather.gov/safety/thunderstorm"> Thunderstorms &amp; Tornadoes</a></li>
        <li><a href="https://www.weather.gov/safety/tornado">Tornado</a></li>
        <li><a href="https://www.weather.gov/safety/tsunami">Tsunami</a></li>
        <li><a href="https://www.weather.gov/safety/wildfire">Wildfire</a></li>
        <li><a href="https://www.weather.gov/safety/wind">Wind</a></li>
        <li><a href="https://www.weather.gov/safety/winter">Winter</a></li>
      </ul>
    </div>
  </li>
  ...
```